virtcontainers/fc.go

```diff
@@ -687,16 +683,53 @@ func (fc *firecracker) resizeVCPUs(reqVCPUs uint32) (currentVCPUs uint32, newVCP
 	return 0, 0, nil
 }
 
-// this is used to apply cgroup information on the host. not sure how necessary this
-// is in the first pass.
+// This is used to apply cgroup information on the host.
 //
-// Need to see if there's an easy way to ask firecracker for thread ids associated with
-// the vCPUs.  Issue opened to ask for per vCPU thread IDs:
-// https://github.com/firecracker-microvm/firecracker/issues/718
-func (fc *firecracker) getThreadIDs() (*threadIDs, error) {
-	//TODO: this may not be exactly supported in Firecracker. Closest is cpu-template as part
-	// of get /machine-config
-	return nil, nil
+// As suggested by https://github.com/firecracker-microvm/firecracker/issues/718,
+// let's use `ps -T -p <pid>` to get fc vcpu info.
+func (fc *firecracker) getThreadIDs() (vcpuThreadIDs, error) {
+	var vcpuInfo vcpuThreadIDs
+
+	psArgs := []string{"-T", "-p", strconv.Itoa(fc.info.PID)}
+	cmd := exec.Command("ps", psArgs...)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return vcpuInfo, errors.Wrapf(err, "fail to get fc thread info: %s", output)
+	}
+
+	// Sample out:
+	//   PID  SPID TTY          TIME CMD
+	// 14328 14328 ?        00:00:00 firecracker
+	// 14328 14330 ?        00:00:00 fc_vmm
+	// 14328 14337 ?        00:00:05 fc_vcpu0
+	lines := strings.Split(string(output), "\n")
```

Comment:
I wonder if won't be easier to read the content of `/proc/[pid]/task/`  :thinking:

Reply:
hmm, I've found `github.com/shirou/gopsutil/process` and will make use of it. Just so that we don't need to fork with `exec.Command`.

Reply:
Turns out `github.com/shirou/gopsutil/process` is also calling `exec.command` so I don't think we should use it.